### PR TITLE
fix(core): Require mfa code to disable mfa

### PR DESCRIPTION
**Pull request by tomi**
## Summary

Continuation of #10341

Also enables rate limiting for the `/mfa/disable` endpoint.

For the FE:
- adds a typed event emitter
- a mechanism to use Modal event emitter to provide data back from the modal

https://github.com/user-attachments/assets/e810dda9-1f30-4ed3-b9ab-c5514eb6ceb3

## Related Linear tickets, Github issues, and Community forum posts

[SEC-67](https://linear.app/n8n/issue/SEC-67/29-account-update-without-mfa)

## Review / Merge checklist

- [ ] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) 
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included. 
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

### CODE REVIEW

File: packages/design-system/src/utils/typed-event-bus.ts
Comment (alexgrozav) [inline packages/design-system/src/utils/typed-event-bus.ts:36]: Great to see you've added event bus types! 

I think we should merge the two event bus types and just set the default generic to `Record<string, unknown>` when creating an event bus. I don't see a reason to keep the two create functions separate.

File: packages/editor-ui/src/components/Modal.vue
Comment (alexgrozav) [inline packages/editor-ui/src/components/Modal.vue:4]: We should avoid defining functions in templates since they get recreated on every render. Create an extra `onCloseDialog` function instead.

**Review by alexgrozav — CHANGES_REQUESTED** [review]
Looks good! I think we should merge the two event bus creator functions though.

File: packages/design-system/src/utils/typed-event-bus.ts
Comment (tomi) [reply packages/design-system/src/utils/typed-event-bus.ts:36]: Makes sense. I kept them separate to have a clear refactoring path to the new one, but we can also combine them already at this point. However, I can't use `Record<string, unknown>` as that will break most of the usage sites because they are missing the correct event types. Will type it with `Record<string, any>` for now.

**Review by alexgrozav — DISMISSED** [review]

File: packages/design-system/src/utils/event-bus.ts
Comment (netroy) [inline packages/design-system/src/utils/event-bus.ts:13]: maybe we should do this in a separate PR?

**Comment by tomi** [comment]
Rebased on top of #10367. Once that is merged we can merge this one as well

**Review by netroy — COMMENTED** [review]
LGTM. will approve after #10367 is merged

**Review by netroy — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.55.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.55.0)

**Pull request merged**
